lib/datapackage/resource.rb

```diff
@@ -2,23 +2,55 @@ module DataPackage
   class Resource < Hash
     include DataPackage::Helpers
 
-    attr_reader :data
+    attr_reader :name, :profile, :source, :source_type, :valid, :errors
 
     def initialize(resource, base_path = '')
       resource = dereference_descriptor(resource, base_path: base_path,
         reference_fields: ['schema', 'dialect'])
-      if resource.fetch('data', nil)
-        @data = resource['data']
-      elsif resource.fetch('path', nil)
-        @data = open(join_paths(base_path, resource['path'])).read
-      else
-        raise ResourceException.new 'A resource descriptor must have a `path` or `data` property.'
-      end
       self.merge! resource
+      @profile = DataPackage::Profile.new(self.fetch('profile', 'data-resource'))
+      @name = self['name']
+      get_source!(base_path)
     end
 
     def table
-      @table ||= TableSchema::Table.new(CSV.parse(data), self['schema']) if self['schema']
+      TableSchema::Table.new(self.source, self['schema']) if tabular?
```

Comment:
Good catch! I fixed it [below](https://github.com/frictionlessdata/datapackage-rb/pull/44/commits/4d71d1f1a06aac0317b747fd1a33fdf9df129a63).